sdk/src/Core/AWSXRayRecorder.netcore.cs

```diff
@@ -203,12 +204,12 @@ public override void BeginSubsegment(string name)
         /// <summary>
         /// Begin a tracing subsegment. A new subsegment will be created and added as a subsegment to previous facade segment or subsegment.
         /// </summary>
-        private void ProcessSubsegmentInLambdaContext(string name)
+        private void ProcessSubsegmentInLambdaContext(string name, DateTime? timestamp = null)
         {
             if (!TraceContext.IsEntityPresent()) // No facade segment available and first subsegment of a subsegment branch needs to be added
             {
                 AddFacadeSegment(name);
-                AddSubsegmentInLambdaContext(new Subsegment(name));
+                AddSubsegmentInLambdaContext(name, timestamp);
```

Comment:
Avoid duplicate code -  `AddSubsegmentInLambdaContext(new Subsegment(name,timestamp))` will do necessary setting of the start time. Same thing for `AddSubsegment()`. In the current function definitions you just remove code line that sets the start time.

Reply:
Currently we don't set properties other than `name` for new `Subsegment` instance.